pkg/webhook/policy.go

```diff
@@ -582,37 +576,41 @@ func (h *validationHandler) reviewRequest(ctx context.Context, req *admission.Re
 		return nil, fmt.Errorf("failed to create augmentedReview: %w", err)
 	}
 
-	// Convert the request's generator resource to unstructured for expansion
-	obj := &unstructured.Unstructured{}
-	if _, _, err := deserializer.Decode(req.Object.Raw, nil, obj); err != nil {
-		return nil, fmt.Errorf("error decoding generator resource %s: %w", req.Name, err)
-	}
-	obj.SetNamespace(req.Namespace)
-	obj.SetGroupVersionKind(
-		schema.GroupVersionKind{
-			Group:   req.Kind.Group,
-			Version: req.Kind.Version,
-			Kind:    req.Kind.Kind,
-		})
-
-	// Expand the generator and apply mutators to the resultant resources
-	// The base object is not mutated, so we do not need to specify its source
-	base := &mutationtypes.Mutable{
-		Object:    obj,
-		Namespace: review.Namespace,
-		Username:  req.AdmissionRequest.UserInfo.Username,
-	}
-	resultants, err := h.expansionSystem.Expand(base)
-	if err != nil {
-		return nil, fmt.Errorf("unable to expand object: %w", err)
+	resultants := []*expansion.Resultant{}
+	// Skip the expansion if admissionRequest.Obj is nil.
+	if req.AdmissionRequest.Object.Raw != nil {
+		// Convert the request's generator resource to unstructured for expansion
+		obj := &unstructured.Unstructured{}
+		if _, _, err := deserializer.Decode(req.Object.Raw, nil, obj); err != nil {
+			return nil, fmt.Errorf("error decoding generator resource %s: %w", req.Name, err)
+		}
+		obj.SetNamespace(req.Namespace)
+		obj.SetGroupVersionKind(
+			schema.GroupVersionKind{
+				Group:   req.Kind.Group,
+				Version: req.Kind.Version,
+				Kind:    req.Kind.Kind,
+			})
+
+		// Expand the generator and apply mutators to the resultant resources
+		// The base object is not mutated, so we do not need to specify its source
+		base := &mutationtypes.Mutable{
+			Object:    obj,
+			Namespace: review.Namespace,
+			Username:  req.AdmissionRequest.UserInfo.Username,
+		}
+		resultants, err = h.expansionSystem.Expand(base)
+		if err != nil {
+			return nil, fmt.Errorf("unable to expand object: %w", err)
+		}
 	}
 
 	trace, dump := h.tracingLevel(ctx, req)
 	resp, err := h.review(ctx, review, trace, dump)
 	if err != nil {
 		return nil, fmt.Errorf("error reviewing resource %s: %w", req.Name, err)
 	}
-
+	// resultants slice will be empty if expand is skipped.
```

Comment:
no need to comment about this, it can be inferred from the code without extra context.

Reply:
Deleted that comment.